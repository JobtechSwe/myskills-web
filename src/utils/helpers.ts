import Cookies from 'js-cookie'
import { LocalStateProps, ContactInformationProps } from '../graphql/client'
import {
  EducationInput,
  Occupation,
  OntologyRelationResponse,
} from '../generated/myskills'

export const getCookie = (name: string) => Cookies.get(name)

export const setCookie = (
  name: string,
  value: string,
  options: object = {}
) => {
  Cookies.set(name, value, options)
}

export const removeCookie = (name: string) => {
  Cookies.remove(name)
}

export const redirect = (route: string) => (location.href = route)

interface StorageEntryProps {
  type: string
}

interface Educations extends StorageEntryProps {
  type: 'educations'
  data: EducationInput[]
}

interface OccupationStorage extends StorageEntryProps {
  type: 'occupation'
  data: Occupation
}

interface Skills extends StorageEntryProps {
  type: 'skills'
  data: OntologyRelationResponse[]
}
interface Traits extends StorageEntryProps {
  type: 'traits'
  data: string[]
}
interface WhoAmI extends StorageEntryProps {
  type: 'whoAmI'
  data: string
}

interface ContactInformation extends StorageEntryProps {
  type: 'contact'
  data: ContactInformationProps
}

export type StorageEntry =
  | ContactInformation
  | Educations
  | OccupationStorage
  | Skills
  | Traits
  | WhoAmI

export const storageHelper = {
  load: (initialState: LocalStateProps) =>
    Object.entries(initialState).reduce(
      (acc, [prop, val]) => ({
        ...acc,
        [prop]: JSON.parse(localStorage.getItem(prop) as string) || val,
      }),
      {} as LocalStateProps
    ),
  set: (payload: StorageEntry) =>
    localStorage.setItem(payload.type, JSON.stringify(payload.data)),
}
