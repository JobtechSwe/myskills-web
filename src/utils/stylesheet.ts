const colors = {
  lassekongo: 'dodgerblue',
  primary: 'pink',
}

const sizes = ['0', '1rem', '2rem', '4rem']

const typography = {
  default: 'Arial',
}

const space = {
  none: sizes[0],
  small: sizes[1],
  medium: sizes[2],
  big: sizes[3],
}

export default { colors, space, typography }
