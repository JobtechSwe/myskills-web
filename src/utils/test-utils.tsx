import * as React from 'react'
import ApolloClient from 'apollo-client'
import { ApolloProvider } from 'react-apollo-hooks'
import { InMemoryCache } from 'apollo-cache-inmemory'
import { MockLink, MockedProvider } from 'react-apollo/test-utils'
import { ThemeProvider } from 'emotion-theming'
import { render as rtlRender } from 'react-testing-library'
import { theme } from 'theme'

export const render = (children: React.ReactNode, mocks?: any[]) => {
  const client = new ApolloClient({
    cache: new InMemoryCache({ addTypename: true }),
    link: new MockLink(mocks || []),
  })

  return rtlRender(
    <MockedProvider addTypename={true} mocks={mocks}>
      <ApolloProvider client={client}>
        <ThemeProvider theme={theme}>{children}</ThemeProvider>
      </ApolloProvider>
    </MockedProvider>
  )
}
