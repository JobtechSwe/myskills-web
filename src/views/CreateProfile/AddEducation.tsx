import React, { useState } from 'react'
import { useMutation } from 'react-apollo-hooks'
import gql from 'graphql-tag'
import { RouteComponentProps, Link } from '@reach/router'
import Grid from '../../components/Grid'
import Input from '../../components/Input'
import Button from '../../components/Button'
import AddedEducations from '../../components/AddedEducations'
import RegistrationLayout from '../../components/Layout/RegistrationLayout'

export const ADD_EDUCATION_CLIENT = gql`
  mutation addEducationClient($education: EducationInput!) {
    addEducationClient(education: $education) @client {
      programme
      school
      start
      end
    }
  }
`

export const ADD_EDUCATION_API = gql`
  mutation addEducationApi($education: EducationInput!) {
    addEducation(education: $education) {
      programme
      end
      school
      start
    }
  }
`

const AddEducation: React.FC<RouteComponentProps> = () => {
  const initialState = {
    programme: '',
    school: '',
    start: '',
    end: '',
  }
  const addEducationClient = useMutation(ADD_EDUCATION_CLIENT)
  const [education, addEducation] = useState(initialState)

  const handleUpdate = (name: string, value: string) => {
    const updated = {
      ...education,
      [name]: value,
    }

    addEducation(updated)
  }

  const handleSubmit = (event: React.FormEvent<HTMLFormElement>) => {
    event.preventDefault()

    addEducationClient({
      variables: {
        education,
      },
    })

    addEducation(initialState)
  }

  return (
    <RegistrationLayout headerText="UTBILDNING" nextPath="beskriv-dig" step={4}>
      <form onSubmit={handleSubmit}>
        <Grid>
          <AddedEducations />

          <Input
            name="programme"
            onChange={({ target }: React.ChangeEvent<HTMLInputElement>) =>
              handleUpdate('programme', target.value)
            }
            placeholder="Program"
            type="text"
            value={education.programme}
          />

          <Input
            name="school"
            onChange={({ target }: React.ChangeEvent<HTMLInputElement>) =>
              handleUpdate('school', target.value)
            }
            placeholder="Skola"
            type="text"
            value={education.school}
          />

          <Input
            name="start"
            onChange={({ target }: React.ChangeEvent<HTMLInputElement>) =>
              handleUpdate('start', target.value)
            }
            placeholder="Från"
            type="date"
            value={education.start}
          />

          <Input
            name="end"
            onChange={({ target }: React.ChangeEvent<HTMLInputElement>) =>
              handleUpdate('end', target.value)
            }
            placeholder="Till"
            type="date"
            value={education.end}
          />

          <Button type="submit">Lägg till</Button>
        </Grid>
      </form>
    </RegistrationLayout>
  )
}

export default AddEducation
