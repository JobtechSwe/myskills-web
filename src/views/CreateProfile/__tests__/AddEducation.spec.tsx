import * as React from 'react'
import AddEducation from '../AddEducation'
import { render } from '../../../utils/test-utils'
import { wait } from 'react-testing-library'
import h from '../../../components/AddedEducations'
jest.mock(
  '../../../components/AddedEducations',
  () =>
    function mockedComponent() {
      return <div />
    }
)

describe('views/AddEducation', () => {
  it('renders empty result', async () => {
    const { container } = render(<AddEducation />)

    await wait()

    expect(container).toMatchSnapshot()
  })
})
