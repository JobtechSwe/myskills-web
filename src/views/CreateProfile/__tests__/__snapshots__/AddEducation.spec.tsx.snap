// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`views/AddEducation renders empty result 1`] = `
<div>
  <section
    class="layout css-ic5la4"
  >
    <div
      class="css-zzz07x"
    >
      <div
        class="css-1y6f7jl"
      />
      <div
        class="css-1y6f7jl"
      />
      <div
        class="css-1y6f7jl"
      />
      <div
        class="css-1y6f7jl"
      />
      <div
        class="css-1l5oznz"
      />
      <div
        class="css-1l5oznz"
      />
    </div>
    <svg
      fill="none"
      height="20"
      viewBox="0 0 27 20"
      width="27"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M0.395325 9.04442L9.04398 0.395757C9.57166 -0.131919 10.4274 -0.131919 10.9551 0.395757C11.4829 0.923542 11.4829 1.77911 10.9551 2.3069L4.61338 8.64863H24.7111C25.4574 8.64863 26.0625 9.25371 26.0625 9.99999C26.0625 10.7461 25.4574 11.3513 24.7111 11.3513H4.61338L10.9549 17.6931C11.4827 18.2209 11.4827 19.0764 10.9549 19.6042C10.6911 19.8679 10.3452 20 9.99934 20C9.6535 20 9.30766 19.8679 9.04377 19.6042L0.395325 10.9556C-0.13246 10.4278 -0.13246 9.5722 0.395325 9.04442Z"
        fill="black"
      />
    </svg>
    <form>
      <div
        class="css-fhv5oa"
      >
        <div />
        <input
          class="css-90stuz"
          font-family="default"
          font-size="medium"
          name="programme"
          placeholder="Program"
          type="text"
          value=""
        />
        <input
          class="css-90stuz"
          font-family="default"
          font-size="medium"
          name="school"
          placeholder="Skola"
          type="text"
          value=""
        />
        <input
          class="css-90stuz"
          font-family="default"
          font-size="medium"
          name="start"
          placeholder="Från"
          type="date"
          value=""
        />
        <input
          class="css-90stuz"
          font-family="default"
          font-size="medium"
          name="end"
          placeholder="Till"
          type="date"
          value=""
        />
        <button
          class="css-199cpun"
          font-family="default"
          font-size="medium"
          type="submit"
        >
          Lägg till
        </button>
      </div>
    </form>
    <div
      class="css-52ud0e"
      style="border: 1px solid black;"
    >
      <button
        class="css-199cpun"
        font-family="default"
        font-size="medium"
      >
        Fortsätt
      </button>
    </div>
  </section>
</div>
`;
